app/main.py

```diff
@@ -1 +1,41 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, health=100, name="", hidden=False):
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self):
+        return f"{{Name: {self.name}, Health: {self.health}, Hidden: {self.hidden}}}"
+
+    def decrease_health(self, amount: int):
+        self.health -= amount
+        if self.health <= 0:
+            self.health = 0
+            self.die()
+
+    def die(self):
+        Animal.alive.remove(self)
+
+
+class Herbivore(Animal):
+    def __init__(self, name):
+        super().__init__(name=name)
```

Comment:
There is no need to duplicate `__init__` in this class and in the `Carnivore` class as well, remove them and rely on the parent class implementation